00_Base/src/broadcaster/session.broadcaster.ts

```diff
@@ -22,34 +31,102 @@ export class SessionBroadcaster extends BaseBroadcaster {
     readonly credentialsService: CredentialsService,
   ) {
     super();
-    this.transactionRepository.transaction.on('created', (transactions) =>
-      this.broadcast(transactions),
-    );
-    this.transactionRepository.transaction.on('updated', (transactions) =>
-      this.broadcast(transactions),
+    this.transactionRepository.transaction.on('created', (transactions) => {
+      if (transactions && transactions.length > 0) {
+        this.logger.info(
+          'Attempting to broadcast created transactions',
+          transactions.map((t) => t.id),
+        );
+        this.broadcast(transactions).then();
+      }
+    });
+    this.transactionRepository.on(
+      'created',
```

Comment:
should this be 'updated' instead?

Reply:
interesting that you point that out because that was my first thought as well - but it seems that the way we have it is that there is one `Transaction` which has multiple `TransactionEvents`, and we use the `TransactionEventRepository` to access those transaction events. The partially confusing part is that the `TransactionEventRepository` has `transaction: TransactionRepository` on it, which is why here we have to subscribe to the creation of the TransactionEvents that come in because they don't really seem to get updated

Reply:
@elliot-sabitov 

You're saying that this is subscribing to creation of TransactionEvents?

Maybe we should consolidate these into methods: `subscribeToTransactionCreated()` and `subscribeToTransactionEventCreated()`?

Reply:
adjusted